src/App.tsx

```diff
@@ -1,14 +1,55 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
+export enum Filters {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [loading, setLoading] = useState(false);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState(Filters.All);
+
+  useEffect(() => {
+    setLoading(true);
+
+    getTodos()
+      .then(setTodos)
+      .finally(() => setLoading(false));
+  }, []);
+
+  let filteredTodos = [...todos];
+
+  switch (filter) {
+    case Filters.Active:
+      filteredTodos = todos.filter(todo => !todo.completed);
+      break;
+
+    case Filters.Completed:
+      filteredTodos = todos.filter(todo => todo.completed);
+      break;
+
+    default:
+      break;
+  }
+
+  filteredTodos = filteredTodos.filter(todo =>
+    todo.title.trim().toLowerCase().includes(query.trim().toLowerCase()),
+  );
```

Comment:
if you haves query and comleted/active filter you are filtering todos twice, in my opinion better option its create function where you will filter only 1 time your todos, its will be better for performance